golang/vaultclient/createAccount.go

```diff
@@ -0,0 +1,127 @@
+package vaultclient
+
+import (
+	"github.com/aws/aws-sdk-go/aws/awsutil"
+	"github.com/aws/aws-sdk-go/aws/request"
+)
+
+import "time"
+
+const opCreateAccount = "CreateAccount"
+
+type CreateAccountInput struct {
+	Name     *string `locationName:"name"`
+	Email    *string `locationName:"emailAddress"`
+	QuotaMax *int64  `locationName:"quotaMax"`
+}
+
+// String returns the string representation
+func (s CreateAccountInput) String() string {
+	return awsutil.Prettify(s)
+}
+
+// Validate inspects the fields of the type to determine if they are valid.
+func (s *CreateAccountInput) Validate() error {
+	invalidParams := request.ErrInvalidParams{Context: "CreateAccountInput"}
+
+	if s.Name == nil {
+		invalidParams.Add(request.NewErrParamRequired("Name"))
+	}
+	if s.Name != nil && len(*s.Name) < 1 {
+		invalidParams.Add(request.NewErrParamMinLen("Name", 1))
+	}
+
+	if s.Email == nil {
+		invalidParams.Add(request.NewErrParamRequired("Email"))
+	}
+	if s.Email != nil && len(*s.Email) < 1 {
+		invalidParams.Add(request.NewErrParamMinLen("Email", 1))
+	}
+
+	if s.QuotaMax != nil && *s.QuotaMax < 1 {
```

Comment:
QuotaMax is not a required value. Accounts can be created without specifying it.

Reply:
also you might need an (optional) option of specifying accountID: https://github.com/scality/vaultclient/blob/development/8.2/lib/IAMClient.js#L125
https://github.com/scality/vaultclient/blob/development/8.2/bin/vaultclient#L107

Reply:
Yes, QuotaMax is not required. I do not check for it.